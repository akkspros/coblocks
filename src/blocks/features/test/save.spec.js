/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { registerBlockType, createBlock, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let serializedBlock;

describe( name, () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );

		// Reset the reused variables.
		serializedBlock = '';
	} );

	it( 'should render', () => {
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render heading h3', () => {
		block.attributes.headingLevel = 3;
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '{\"headingLevel\":3}' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render gutter 3 columns', () => {
		block.attributes.columns = 3;
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'has-3-columns' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render gutter large', () => {
		block.attributes.gutter = 'large';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'has-large-gutter' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render padding large', () => {
		block.attributes.paddingSize = 'large';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'has-large-padding' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render margin huge', () => {
		block.attributes.marginSize = 'huge';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'has-huge-margin' );
		expect( serializedBlock ).toMatchSnapshot();
	} );
} );
