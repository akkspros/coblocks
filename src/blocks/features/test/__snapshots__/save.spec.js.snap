// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/features should render 1`] = `
"<!-- wp:coblocks/features -->
<div class=\\"wp-block-coblocks-features\\"><div class=\\"wp-block-coblocks-features__inner has-columns has-2-columns has-responsive-columns has-medium-gutter has-center-content\\"></div></div>
<!-- /wp:coblocks/features -->"
`;

exports[`coblocks/features should render gutter 3 columns 1`] = `
"<!-- wp:coblocks/features {\\"columns\\":3} -->
<div class=\\"wp-block-coblocks-features\\"><div class=\\"wp-block-coblocks-features__inner has-columns has-3-columns has-responsive-columns has-medium-gutter has-center-content\\"></div></div>
<!-- /wp:coblocks/features -->"
`;

exports[`coblocks/features should render gutter large 1`] = `
"<!-- wp:coblocks/features {\\"gutter\\":\\"large\\"} -->
<div class=\\"wp-block-coblocks-features\\"><div class=\\"wp-block-coblocks-features__inner has-columns has-2-columns has-responsive-columns has-large-gutter has-center-content\\"></div></div>
<!-- /wp:coblocks/features -->"
`;

exports[`coblocks/features should render heading h3 1`] = `
"<!-- wp:coblocks/features {\\"headingLevel\\":3} -->
<div class=\\"wp-block-coblocks-features\\"><div class=\\"wp-block-coblocks-features__inner has-columns has-2-columns has-responsive-columns has-medium-gutter has-center-content\\"></div></div>
<!-- /wp:coblocks/features -->"
`;

exports[`coblocks/features should render margin huge 1`] = `
"<!-- wp:coblocks/features {\\"marginSize\\":\\"huge\\"} -->
<div class=\\"wp-block-coblocks-features\\"><div class=\\"wp-block-coblocks-features__inner has-columns has-2-columns has-responsive-columns has-medium-gutter has-margin has-huge-margin has-center-content\\"></div></div>
<!-- /wp:coblocks/features -->"
`;

exports[`coblocks/features should render padding large 1`] = `
"<!-- wp:coblocks/features {\\"paddingSize\\":\\"large\\"} -->
<div class=\\"wp-block-coblocks-features\\"><div class=\\"wp-block-coblocks-features__inner has-columns has-2-columns has-responsive-columns has-medium-gutter has-padding has-large-padding has-center-content\\"></div></div>
<!-- /wp:coblocks/features -->"
`;
